src/components/organisms/CesiumViewer.vue

```diff
@@ -299,35 +302,108 @@ export default defineComponent({
     });
 
     watch(specifiedShape, () => {
-      const uploadedFootPrint: any[] = [];
+      const uploadedFootprint: any[] = [];
       specifiedShape.value.coordinates[0].forEach((e: any) => {
-        uploadedFootPrint.push(Cesium.Cartesian3.fromDegrees(e[0], e[1]));
+        uploadedFootprint.push(Cesium.Cartesian3.fromDegrees(e[0], e[1]));
       });
       cesiumViewer.value.entities.add({
         polygon: {
-          hierarchy: uploadedFootPrint,
+          hierarchy: uploadedFootprint,
           material: new Cesium.ColorMaterialProperty(
             Cesium.Color.RED,
           ),
         },
       });
     }, { deep: true });
 
-    watch(footPrints, () => {
+    const addGeojson = (geojson: { coordinates: any[][] }) => {
+      const cesiumPoints: RGDResult[] = [];
+      geojson.coordinates[0].forEach((e: any) => {
+        cesiumPoints.push(Cesium.Cartesian3.fromDegrees(e[0], e[1]));
+      });
+      return cesiumViewer.value.entities.add({
+        polygon: {
+          hierarchy: cesiumPoints,
+          material: new Cesium.ColorMaterialProperty(
+            Cesium.Color.fromRandom({ alpha: 0.5 }),
+          ),
+        },
+      });
+    };
+
+    const footprintEntities: any = {}; // Cesium.Entity
+    const addFootprint = async (spatialId: number) => {
+      if (!(spatialId in footprintEntities)) {
+        const element = await rgdFootprint(spatialId);
+        footprintEntities[spatialId] = addGeojson(element.footprint);
+      }
+    };
+    const removeFootprint = (spatialId: number) => {
+      if (spatialId in footprintEntities) {
+        cesiumViewer.value.entities.remove(footprintEntities[spatialId]);
+        delete footprintEntities[spatialId];
+      }
+    };
+    const updateFootprints = () => {
+      // Purge footprints
+      Object.keys(footprintEntities).forEach((key: string) => {
+        const spatialId = Number(key);
+        if (footprintIds.value.indexOf(spatialId) < 0) {
+          removeFootprint(spatialId);
+        }
+      });
+
+      // Add footprints
+      // eslint-disable-next-line no-unused-expressions
+      footprintIds.value?.forEach((spatialId: number) => {
+        addFootprint(spatialId);
+      });
+    };
+
+    watch(footprintIds, updateFootprints, { deep: true });
+
+    // TODO: this logic needs to be moved to a better place, IMO the active element view as data types are checked
+    const generateTileProvider = async (imageId: number, band = 0) => {
+      const data = await rgdImageTilesMeta(imageId);
+      const extents = data.bounds;
+      const rectangle = Cesium.Rectangle.fromDegrees(extents.xmin, extents.ymin, extents.xmax, extents.ymax);
+      const tileProvider = new Cesium.UrlTemplateImageryProvider({
+        // TODO: how do we do this with Axios?
+        url: `http://localhost:8000/api/image_process/imagery/${imageId}/tiles/{z}/{x}/{y}.png?projection=EPSG:3857&band=${band}`,
```

Comment:
@marySalvi and @annehaley, how can we generate this URL from the Axios instance?